### ISSUE TITLE
LoadError: InitError: Evaluation into the closed module `GR` breaks incremental compilation

### ISSUE THREAD

**Issue by mkitti** [jheinen/GR.jl#390]
When I break GR v0.57.3 by deleting `GR.depsfile`:

```julia
julia> using GR

julia> rm(GR.depsfile)
```

and then try to use `Plots.jl` in a **new** Julia session, I get the following:

```julia
julia> using Plots
[ Info: Precompiling Plots [91a5bcdd-55d7-5caf-9e0b-520d859cae80]
Your GR installation is incomplete. Rerunning build step for GR package.
ERROR: LoadError: LoadError: InitError: Evaluation into the closed module `GR` breaks incremental compilation because the side effects will not be permanent. This is likely due to some other module mutating `GR` with `eval` during precompilation - don't do this.
Stacktrace:
  [1] eval
    @ .\boot.jl:360 [inlined]
  [2] __init__()
    @ GR ~\.julia\packages\GR\qSS8e\src\GR.jl:311
  [3] _include_from_serialized(path::String, depmods::Vector{Any})
    @ Base .\loading.jl:674
  [4] _require_from_serialized(path::String)
    @ Base .\loading.jl:727
  [5] _require(pkg::Base.PkgId)
    @ Base .\loading.jl:1031
  [6] require(uuidkey::Base.PkgId)
    @ Base .\loading.jl:914
  [7] require(into::Module, mod::Symbol)
    @ Base .\loading.jl:901
  [8] include(mod::Module, _path::String)
    @ Base .\Base.jl:386
  [9] include(x::String)
    @ Plots ~\.julia\packages\Plots\SVksJ\src\Plots.jl:1
 [10] top-level scope
    @ ~\.julia\packages\Plots\SVksJ\src\Plots.jl:218
 [11] include
    @ .\Base.jl:386 [inlined]
 [12] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base .\loading.jl:1213
 [13] top-level scope
    @ none:1
 [14] eval
    @ .\boot.jl:360 [inlined]
 [15] eval(x::Expr)
    @ Base.MainInclude .\client.jl:446
 [16] top-level scope
    @ none:1
during initialization of module GR
in expression starting at C:\Users\kittisopikulm\.julia\packages\Plots\SVksJ\src\backends\gr.jl:6
in expression starting at C:\Users\kittisopikulm\.julia\packages\Plots\SVksJ\src\Plots.jl:1
ERROR: Failed to precompile Plots [91a5bcdd-55d7-5caf-9e0b-520d859cae80] to C:\Users\kittisopikulm\.julia\compiled\v1.6\Plots\jl_DD56.tmp.
Stacktrace:
 [1] error(s::String)
   @ Base .\error.jl:33
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::Base.TTY, internal_stdout::Base.TTY)
   @ Base .\loading.jl:1360
 [3] compilecache(pkg::Base.PkgId, path::String)
   @ Base .\loading.jl:1306
 [4] _require(pkg::Base.PkgId)
   @ Base .\loading.jl:1021
 [5] require(uuidkey::Base.PkgId)
   @ Base .\loading.jl:914
 [6] require(into::Module, mod::Symbol)
   @ Base .\loading.jl:901
```

The initial work around is just to try the command again in the same session:

```julia
julia> using Plots
Your GR installation is incomplete. Rerunning build step for GR package.
    Building GR → `C:\Users\kittisopikulm\.julia\scratchspaces\44cfe95a-1eb2-52ea-b672-e2afdf69b78f\82a03d9c331d181bf33c99c9af04202cc4533d48\build.log`
  Progress [========================================>]  11/11
  ? GRUtils
  ? Plots
9 dependencies successfully precompiled in 14 seconds (221 already precompiled)
2 dependencies failed but may be precompilable after restarting julia
[ Info: GR was successfully rebuilt
[ Info: Precompiling Plots [91a5bcdd-55d7-5caf-9e0b-520d859cae80]
┌ Warning: Module GR with build ID 2425027140852901 is missing from the cache.
│ This may mean GR [28b8d3ca-fb5f-59d9-8090-bfdbd6d07a71] does not support precompilation but is imported by a module that does.
└ @ Base loading.jl:1008
[ Info: Skipping precompilation since __precompile__(false). Importing Plots [91a5bcdd-55d7-5caf-9e0b-520d859cae80].

julia> x = [0, 0.2, 0.4, 0.6, 0.8, 1.0]; y = [0.3, 0.5, 0.4, 0.2, 0.6, 0.7]
6-element Vector{Float64}:
 0.3
 0.5
 0.4
 0.2
 0.6
 0.7

julia> plot(x,y)
```

This will produce a plot.

**Comment by mkitti** [comment]
If you start a new session, after this you will get normal precompilation:

```julia
julia> using Plots
[ Info: Precompiling Plots [91a5bcdd-55d7-5caf-9e0b-520d859cae80]

julia> x = [0, 0.2, 0.4, 0.6, 0.8, 1.0]; y = [0.3, 0.5, 0.4, 0.2, 0.6, 0.7]
6-element Vector{Float64}:
 0.3
 0.5
 0.4
 0.2
 0.6
 0.7

julia> plot(x,y)
```

**Comment by mkitti** [comment]
The cause of the error upon `using Plots` is that GR's `__init__` currently evals code into the GR module:

https://github.com/jheinen/GR.jl/blob/master/src/GR.jl#L311-L316

The first part invokes `Pkg.build()`. This does not need to be executed within `GR`. Additionally it could be done by just executing [deps/build.jl](deps/build.jl).

The second part of it includes `deps.jl` which might contain `using GR_jll`. For this we might be able to avoid that by using Requires.jl to dynamically load `GR_jll` if available.

**Comment by mkitti** [comment]
@jheinen The incremental compilation issue should be closed by #395 

```julia
julia> using GR

julia> rm(GR.depsfile)
```

In a new Julia session:
```julia
julia> using Plots # No precompilation needed

julia> ccall( (:gr_initgr, "libGR",), Nothing, () ) # GR shared libraries are not loaded yet, no possible interference with other backends
ERROR: could not load library "libGR"
The specified module could not be found.
Stacktrace:
 [1] top-level scope
   @ .\REPL[2]:1

julia> x = [0, 0.2, 0.4, 0.6, 0.8, 1.0]; y = [0.3, 0.5, 0.4, 0.2, 0.6, 0.7]
6-element Vector{Float64}:
 0.3
 0.5
 0.4
 0.2
 0.6
 0.7

julia> plot(x,y) # GR shared libraries are loaded now, just in time

julia> ccall( (:gr_initgr, "libGR",), Nothing, () ) # GR shared libraries are loaded now, so we can call GR framework functions

```
